database/src/main/postgres/runs/_validate_partitioning.sql

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION runs._validate_partitioning(
+    IN i_partitioning JSONB,
+    IN i_is_pattern BOOL = false
+) RETURNS SETOF TEXT AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: runs._validate_partitioning(2)
+--      Validates the input partitioning and returns a set of individual error messages, one per an issue.
+--      The validation performs:
+--          1) Correct structure of the input JSONB object
+--          2) The list of keys in 'keys' is unique and doesn't have NULLs
+--          3) The keys in 'keys' and in 'keysToValuesMap' correspond to each other
+--          4) (if i_is_pattern = false) The values in 'keysToValuesMap' are non-empty/non-null
+--
+-- Parameters:
+--      i_partitioning      - partitioning to validate, a valid example:
+-- 	                          {
+--                              "keys": ["one", "two", "three"],
+--                              "version": 1,
+-- 	                            "keysToValuesMap": {
```

Comment:
**Question:**
Wouldn't `values be enough?
```suggestion
-- 	                        "values": {
```

Reply:
maybe `keysToValues`

Reply:
changed